### DISCUSSION TITLE
sqlite.SqliteException@src/sqlite.d(128): out of memory

### DISCUSSION THREAD

**Discussion by matteioo** [abraunegg/onedrive#2289] [Q&A] [answered]
Hello,
I am using fedora 37 which I have reinstalled yesterday. Now I wanted to reinstall your program (which previously worked perfectly) but now I only get the following message after typing `onedrive --resync` in the terminal:

```
[user@fedora ~]$ onedrive --resync
Configuration file successfully loaded

The use of --resync will remove your local 'onedrive' client state, thus no record will exist regarding your current 'sync status'
This has the potential to overwrite local versions of files with potentially older versions downloaded from OneDrive which can lead to data loss
If in-doubt, backup your local data first before proceeding with --resync

Are you sure you wish to proceed with --resync? [Y/N] y

Deleting the saved application sync status ...

WARNING: Your onedrive client version is obsolete and unsupported. Please upgrade your client version.

Configuring Global Azure AD Endpoints

--synchronize or --monitor switches missing from your command line input. Please add one (not both) of these switches to your command line or use 'onedrive --help' for further assistance.

No OneDrive sync will be performed without one of these two arguments being present.

sqlite.SqliteException@src/sqlite.d(128): out of memory
----------------
??:? [0x4cae9d]
??:? [0x4f29d3]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7fdb6fdfcbab]
??:? _d_run_main2 [0x7fdb6fdfc9c5]
??:? _d_run_main [0x7fdb6fdfc81d]
??:? [0x7fdb6f56a50f]
??:? __libc_start_main [0x7fdb6f56a5c8]
??:? [0x476df4]
```

The destination directory exists (default path: `~/Onedrive/`) and there is a `config` and a `sync_list` (bot in the default config-directory `~/.config/onedrive/`) which got successfully detected and also read when outputting the configs. 

Unfortunately I cannot find a solution to this problem.

**Comment by matteioo** [comment]
Note: I am using version 2.4.21 installed today according to the [installation guide](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md).

**Comment by abraunegg** [comment, marked as answer]
@M00nD3v 
Unfortunately you are running an obsolete and unsupported client version because the Fedora package maintainer is not keeping the Fedora packages current.

Your particular issue is most likely fixed with https://github.com/abraunegg/onedrive/pull/2244 which was included in the v2.4.22 release, and we are now at v2.4.23.

Your options to resolve:
* Email the Fedora package maintainer and get them to update the Fedora packages. Refer to:
  * https://koji.fedoraproject.org/koji/packageinfo?packageID=26044
  * https://bugzilla.redhat.com/buglist.cgi?quicksearch=onedrive
* Build the v2.4.23 RPM yourself. Refer to:
  * https://github.com/abraunegg/onedrive/blob/master/docs/build-rpm-howto.md
* Build the application from source
* Live with it until the Fedora packages get updated

**Reply by matteioo** [reply]
Interestingly it worked previously (like couple days ago), because I always wondered why there was no update whilst it displayed the new version every time I run the command.

**Reply by matteioo** [reply]
Unfortunately I am new to this and I don't know how to get in contact with the fedora team (or is it someone else) in order to build the latest version.

**Reply by abraunegg** [reply]
> Interestingly it worked previously, because I always wondered why there was no update whilst it displayed the new version every time I run the command.

I cant comment as to why .. however `onedrive --resync` is incorrect use and triggers that particular bug .. correct use is:
```
onedrive --synchronize --resync
```
or
```
onedrive --monitor --resync
```

There could be any number of reasons why it previously worked - the differences in compiler (LDC) or sqlite between your prior OS version and Fedora 37 the most likely suspect, where previously it was gracefully handling an error - that later versions in Fedora 37 are more strict about which necessitated the fix in https://github.com/abraunegg/onedrive/pull/2244 which resolves the issue.

**Reply by abraunegg** [reply]
> Unfortunately I am new to this and I don't know how to get in contact with the fedora team (or is it someone else) in order to build the latest version.

If you create an account on Bugzilla, you can add a comment into the ticket about the new client available, from there you can also get the package maintainers email address and email them directly.

**Reply by matteioo** [reply]
The issue is probably, that I tried running the command with both the `--resync` and `--dry-run` flag. By running only `onedrive --synchronize --resync` it is synchronizing. At least part of the folders, so I guess I've messed up the `sync_dir` file.

**Comment by abraunegg** [comment]
@M00nD3v 
A follow up here is that all Fedora packages are now at v2.4.23
